File: src/main/java/at/rueckgr/spotlight/service/Worker.java
package at.rueckgr.spotlight.service;

import at.rueckgr.spotlight.model.DownloadableImage;
import org.apache.logging.log4j.Logger;


public class Worker implements Runnable {

    private static final Logger log = org.apache.logging.log4j.LogManager.getLogger(Worker.class);
    private final ImageDataDownloaderService imageDataDownloaderService;
    private final DownloadDirectoryService downloadDirectoryService;

    public Worker(ImageDataDownloaderService imageDataDownloaderService, DownloadDirectoryService downloadDirectoryService) {
        this.imageDataDownloaderService = imageDataDownloaderService;
        this.downloadDirectoryService = downloadDirectoryService;
    }

    public void run() {
        final DownloadableImage downloadableImage = imageDataDownloaderService.fetchImageData();
        if (downloadableImage == null) {
            log.info("No image found, aborting");
            return;
        }

        downloadDirectoryService.saveImageIfNotExists(downloadableImage);
    }
}

File: src/main/kotlin/at/rueckgr/spotlight/service/Worker.kt
package at.rueckgr.spotlight.service

import at.rueckgr.spotlight.model.DownloadableImage
import at.rueckgr.spotlight.util.Logging
import at.rueckgr.spotlight.util.SpotlightException
import at.rueckgr.spotlight.util.logger

class Worker (private val imageDataDownloaderService: ImageDataDownloaderService,
              private val downloadDirectoryService: DownloadDirectoryService): Logging {
    fun run() {
        val downloadableImage: DownloadableImage? = imageDataDownloaderService.fetchImageData()
        if (downloadableImage == null) {
            logger().info("No image found, aborting")
            return
        }

        try {
            downloadDirectoryService.saveImageIfNotExists(downloadableImage)
        }
        catch (e: SpotlightException) {
            logger().error("Failed to save image", e)
        }
    }
}
